### DISCUSSION TITLE
v3 failed

### DISCUSSION THREAD

**Discussion by wlc002** [acheong08/ChatGPT#986] [General]
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='api.openai.com', port=443): Max retries exceeded with url: /v1/chat/completions (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f05c90c2590>: Failed to establish a new connection: [Errno 101] Network is unreachable'))

v3 doset' work, does someone have this return?

**Comment by acheong08** [comment]
Blocked in your country?

**Reply by wlc002** [reply]
in china
have any soloutions? like proxy?

**Reply by acheong08** [reply]
yes. proxy or vpn

**Reply by wlc002** [reply]
it work whole day now it can't work

**Reply by wlc002** [reply]
how to set up a proxy ,any instructor? heip me ,Thank u

**Comment by wlc002** [comment]
curl https://api.openai.com/v1/completions   -H 'Content-Type: application/json'   -H 'Authorization: Bearer sk-oY7tZJRvV28fTnrKR2qjT3BlbkFJxQ6EBCT8BOKOcJbG8'   -d '{
  "model": "text-davinci-003",
  "prompt": "请介绍一下你自己",
  "max_tokens": 7,
  "temperature": 0
}'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:21 --:--:--     0
curl: (28) Failed to connect to api.openai.com port 443 after 21372 ms: Couldn't connect to server

i use vpn local，but still get 443

**Reply by hongyi-zhao** [reply]
See [here](https://github.com/acheong08/ChatGPT/issues/987) for more details on proxy setting.

Curl based usage is rather simpler, as shown below:

```shell
werner@X10DAi:~$ curl -x socks5h://127.0.0.1:18887 https://api.openai.com/v1/completions -H 'Content-Type: application/json'  -H 'Authorization: Bearer sk-xxx' -d '{
"model": "text-davinci-003",
"prompt": "请介绍一下你自己",
"max_tokens": 7,
"temperature": 0
}'
{"id":"cmpl-6pmlDZTBVAMt97D5hKa4cctXMx6qZ","object":"text_completion","created":1677800183,"model":"text-davinci-003","choices":[{"text":"\n\n我叫XXX","index":0,"logprobs":null,"finish_reason":"length"}],"usage":{"prompt_tokens":17,"completion_tokens":7,"total_tokens":24}}
```
